presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcTableHandle.java

```diff
@@ -51,13 +55,15 @@ public JdbcTableHandle(
             @JsonProperty("catalogName") @Nullable String catalogName,
             @JsonProperty("schemaName") @Nullable String schemaName,
             @JsonProperty("tableName") String tableName,
+            @JsonProperty("columns") Optional<List<JdbcColumnHandle>> columns,
             @JsonProperty("constraint") TupleDomain<ColumnHandle> constraint,
             @JsonProperty("limit") OptionalLong limit)
     {
         this.schemaTableName = requireNonNull(schemaTableName, "schemaTableName is null");
         this.catalogName = catalogName;
         this.schemaName = schemaName;
         this.tableName = requireNonNull(tableName, "tableName is null");
+        this.columns = columns.map(ImmutableList::copyOf);
```

Comment:
rnn `columns`

Reply:
It’s unnecessary. The call to copy the list will fail eagerly with an NPE anyway. There’s no danger of holding on to an invalid value that will later cause the program to fail - this is the actual goal of validating arguments eagerly.

Reply:
yes, but it would fail with NPE without clear message, so we still prefer rnn

Reply:
You look at the stacktrace see it (since you're going to be debugging anyway). The original motivation for adding these when we started using them in the early years of airlift was what I described above, but ok. One day when we move to Java 14+ we can get rid of all these usages.